Syntax errors like "EOL while scanning string literal" in test files

@claude 

The main failure in this job is caused by the code formatter (Black) being unable to parse several test files due to syntax errors, specifically "EOL while scanning string literal", and possibly the use of newer Python syntax unsupported by the runner's Python version.

Key issues found in the logs:
- Syntax errors like "EOL while scanning string literal" in test files (e.g., tests/test_file_analyzer.py, tests/test_database.py, tests/test_scanner.py).
- Black fails to format these files due to these parsing errors.
- The error message hints at running Black with an older Python version that does not support new syntax.

To resolve this:

1. **Fix the Syntax Errors in Test Files**
   - Open the reported files and fix any unterminated string literals or any other syntax errors.
     - For example, make sure all strings are properly closed with matching quotes.
     - Example from the error: line 153 in tests/test_file_analyzer.py, line 254 in tests/test_database.py, line 532 in tests/test_scanner.py.
   - You can search for lines with odd or missing quote marks.

   Example fix:
   ```python
   # Incorrect (unterminated string)
   some_text = "This line is missing the closing quote

   # Correct
   some_text = "This line now has the closing quote"
   ```

2. **Ensure Your Runner Uses the Correct Python Version**
   - If your code uses syntax only supported in Python 3.10+, ensure the CI runner uses the proper Python version.
   - In your `.github/workflows/ci.yml`, explicitly set the Python version:

   ```yaml
   - uses: actions/setup-python@v4
     with:
       python-version: '3.11'
   ```

3. **After Fixing, Re-run the CI**
   - Commit the syntax fixes and any workflow changes.
   - Push to trigger the workflow again.

Summary:
- Fix all unterminated string and syntax errors in the files reported by Black.
- Ensure the CI runner uses a Python version compatible with your code's syntax.
- Re-run CI to verify the job now passes.

Oh no! 💥 💔 💥
81 files would be reformatted, 124 files would be left unchanged, 10 files would fail to reformat.
Error: Process completed with exit code 123.

--- a/tests/test_database.py
+++ b/tests/test_database.py
@@ -251,8 +251,7 @@ class TestDatabasePerformance:
 
         # Print results
         print(
-            f"Single insert time: {
-                single_time:.6f}s for {single_inserts} inserts"
+            f"Single insert time: {single_time:.6f}s for {single_inserts} inserts"
         )
         print(f"Bulk insert time: {bulk_time:.6f}s for {bulk_size} inserts")
         print(f"Speedup: {speedup:.2f}x")
@@ -321,16 +320,12 @@ class TestDatabasePerformance:
         for result in query_times:
             print(f"Database size: {result['size']} alerts")
             print(
-                f"  All alerts query: {
-                    result['all_time']:.6f}s ({
-                    result['all_per_row'] *
-                    1000:.2f}ms per row)"
+                f"  All alerts query: {result['all_time']:.6f}s "
+                f"({result['all_per_row'] * 1000:.2f}ms per row)"
             )
             print(
-                f"  Filtered query: {
-                    result['filter_time']:.6f}s ({
-                    result['filter_per_row'] *
-                    1000:.2f}ms per row)"
+                f"  Filtered query: {result['filter_time']:.6f}s "
+                f"({result['filter_per_row'] * 1000:.2f}ms per row)"
             )
 
         # Verify performance is acceptable

--- a/tests/test_file_analyzer.py
+++ b/tests/test_file_analyzer.py
@@ -150,9 +150,8 @@ class TestFileAnalyzerPerformance:
         # Calculate improvement percentage
         improvement = (1 - (cached_time / uncached_time)) * 100
         print(
-            f"LRU Cache Performance: Uncached: {
-                uncached_time:.6f}s, Cached: {
-                cached_time:.6f}s"
+            f"LRU Cache Performance: Uncached: {uncached_time:.6f}s, "
+            f"Cached: {cached_time:.6f}s"
         )
         print(f"Performance improvement: {improvement:.2f}%")
 
@@ -172,9 +171,8 @@ class TestFileAnalyzerPerformance:
         # Calculate per-file time
         per_file_time = duration / (5 * len(test_files["files"]))
         print(
-            f"Metadata extraction: {
-                duration:.6f}s total, {
-                per_file_time:.6f}s per file"
+            f"Metadata extraction: {duration:.6f}s total, "
+            f"{per_file_time:.6f}s per file"
         )
 
         # Should be reasonably fast (<50ms per file)

--- a/tests/test_scanner.py
+++ b/tests/test_scanner.py
@@ -529,9 +529,10 @@ class TestScannerPerformance:
                 unfiltered_per_file = unfiltered_time / unfiltered_result["scanned"]
                 assert (
                     filtered_per_file <= unfiltered_per_file * 2.5
-                ), f"Filtered scan ({filtered_per_file *
-                                                1000:.2f}ms/file) should be efficient compared to unfiltered ({unfiltered_per_file *
-                                                                                                               1000:.2f}ms/file)"
+                ), (
+                    f"Filtered scan ({filtered_per_file * 1000:.2f}ms/file) should be efficient "
+                    f"compared to unfiltered ({unfiltered_per_file * 1000:.2f}ms/file)"
+                )
 
         finally:
             # Clean up

